traveler_core_kit/src/main/java/com/guestlogix/travelercorekit/Router.java

```diff
@@ -240,13 +237,13 @@ private RouteBuilder(Context context, String apiKey) {
             this.headers.put("x-region", region);
             this.headers.put("x-application-id", applicationId);
             this.headers.put("x-timezone", "UTC");
-            this.headers.put("x-api-key", apiKey);
-            this.context = context;
+            this.headers.put("x-api-key", session.getApiKey());
+
+            SharedPreferences sharedPreferences = context.getSharedPreferences("TRAVELER_PREFS", Context.MODE_PRIVATE);
+            travelerSDKEndpoint = sharedPreferences.getString("TRAVELER_SDK_ENDPOINT", DEFAULT_ENDPOINT);
```

Comment:
We should not be reading from shared prefs everytime. This is super inefficient!

Reply:
I realize this was being done this way before as well. but we should fix this ASAP.

Hint: Use a lazy loaded static variable